examples/python/s2ag-datasets.py

```diff
@@ -0,0 +1,60 @@
+"""
+Sample usage of Semantic Scholar Academic Graph Datasets API
+https://api.semanticscholar.org/api-docs/datasets
+"""
+import requests
+import urllib
+import json
+import os
+
+# Get info about the latest release
+latest_release = requests.get("http://api.semanticscholar.org/datasets/v1/release/latest").json()
+print(latest_release['README'])
+print(latest_release['release_id'])
+
+# Get info about past releases
+dataset_ids = requests.get("http://api.semanticscholar.org/datasets/v1/release").json()
+earliest_release = requests.get(f"http://api.semanticscholar.org/datasets/v1/release/{dataset_ids[0]}").json()
+
+# Print names of datasets in the release
+print("\n".join(d['name'] for d in latest_release['datasets']))
+
+# Print README for one of the datasets
+print(latest_release['datasets'][2]['README'])
+
+# Get info about the papers dataset
+papers = requests.get("http://api.semanticscholar.org/datasets/v1/release/latest/dataset/papers",headers={'x-api-key':os.getenv("S2_API_KEY")}).json()
```

Comment:
It'd be great if the examples work "out of the box", but the API key is required for some functionality which makes this a bit harder. The best solution I can think of is to import a constants.py file similar to what Shaurya has in https://github.com/shauryr/S2QA/blob/master/notebooks/constants.py and maybe implement a bit of logic there which tries to read os.getenv("S2_API_KEY"), and if it's empty we prompt the user for their key (along with a link to the partner request form). Happy to add this as follow-up work to this PR but wanted to pick your brain about how we achieve the goal of making the examples in the repo run out of the box.

Reply:
Agree they should work out of the box. Since they're just examples, I think it's okay to check for a fixed env var and make an unauthenticated request if it's not there. I'll make that change.

Prompting the user for a key is overkill for a sample script, IMO. Might be useful in a client library.

Reply:
Great, thank you!